pageserver/src/walingest.rs

```diff
@@ -1811,11 +1856,36 @@ mod tests {
         // TODO
     }
 
-    static ZERO_CHECKPOINT: Bytes = Bytes::from_static(&[0u8; SIZEOF_CHECKPOINT]);
+    const fn max_of(elements: &[usize]) -> usize {
+        let mut res = elements[0];
+        let mut elem = 1;
+        loop {
+            if elem >= elements.len() {
+                break;
+            }
+            if res < elements[elem] {
+                res = elements[elem];
+            }
+            elem += 1;
+        }
+        res
+    }
+
+    static ZERO_CHECKPOINT: Bytes = Bytes::from_static(
+        &[0u8; max_of(&[
+            postgres_ffi::v14::xlog_utils::SIZEOF_CHECKPOINT,
+            postgres_ffi::v15::xlog_utils::SIZEOF_CHECKPOINT,
+            postgres_ffi::v16::xlog_utils::SIZEOF_CHECKPOINT,
+        ])],
+    );
 
     async fn init_walingest_test(tline: &Timeline, ctx: &RequestContext) -> Result<WalIngest> {
         let mut m = tline.begin_modification(Lsn(0x10));
-        m.put_checkpoint(ZERO_CHECKPOINT.clone())?;
+        dispatch_pgversion!(tline.pg_version, {
+            m.put_checkpoint(Bytes::from_static(
+                &ZERO_CHECKPOINT[..pgv::xlog_utils::SIZEOF_CHECKPOINT],
+            ))?;
```

Comment:
This `max_of` + slicing business is needless micro-optimization.

Please move ZERO_CHECKPOINT into each `vXX` module and in the test just do `pgv::ZERO_CHECKPOINT`.

Also, do we have a test somewhere that checks that `pgv::CheckPoint` can deserialize the `pgv::ZERO_CHECKPOINT`?
If not please add one?